### ISSUE TITLE
Filtering by coordinate system that doesn't exist returns an empty SpatialData object

### ISSUE THREAD

**Issue by rahulbshrestha** [scverse/spatialdata#239]
If a SpatialData object contains the following:
<img width="683" alt="image" src="https://user-images.githubusercontent.com/22676591/236182032-4552dcf6-5ac7-4bf7-aecc-3e2988ad9c8c.png">

If we filter by an incorrect coordinate system name:
`sdata = sdata.filter_by_coordinate_system("X")`

This returns an empty sdata object.
<img width="290" alt="image" src="https://user-images.githubusercontent.com/22676591/236182209-460d6683-6e3b-4520-a4e3-f4b36466cbc2.png">

An error message would be good to inform that the coordinate system doesn't exist.

**Comment by kevinyamauchi** [comment]
Good point. I would be open to raising an exception. I think `None` would also be an option. I am a little concerned that raising an error would be too harsh and would force libraries depending on SpatialData to first check for a coordinate system before making a query. thoughts, @giovp , @LucaMarconato , @EliHei2 ?

**Comment by LucaMarconato** [comment]
checking that the coordinate system exists can be done easily by checking the property
```
    @property
    def coordinate_systems(self) -> list[str]:
```
of `SpatialData`, so I would be in favor for raising an exception. This is also what we do in other cases, like when we want to transform a sdata object.

**Comment by LucaMarconato** [comment]
> **This whole message is AI-generated.** The issue was automatically discovered and reported by an AI agent (Claude) during an autonomous bug hunt on the `spatialdata` code base. It has **not** been verified or triaged by a human yet; the `needs: triage` label is set so that a maintainer can confirm it. The reproduction script below was executed by the agent in an isolated environment (see *Environment*) and its output is pasted verbatim.

I re-attempted to reproduce this issue on the current `main` branch (commit `ccf1ea048d054b6624214bf618008a9f9ae223e0`) as part of a triage pass over open issues, and confirmed it reproduces.

**Reproduction script** (PEP 723 inline metadata; run with `uv run repro.py`, no other setup needed):

```python
# /// script
# requires-python = ">=3.12"
# dependencies = [
#     "spatialdata @ git+https://github.com/scverse/spatialdata.git@ccf1ea048d054b6624214bf618008a9f9ae223e0",
# ]
# ///
"""Issue #239: `filter_by_coordinate_system()` with a non-existing coordinate system silently returns
an empty SpatialData object instead of raising."""
import numpy as np

from spatialdata import SpatialData
from spatialdata.models import Image2DModel

sdata = SpatialData(images={"img": Image2DModel.parse(np.zeros((1, 8, 8), dtype=np.uint8))})
print("coordinate systems:", sdata.coordinate_systems)
try:
    out = sdata.filter_by_coordinate_system("does_not_exist")
    print("filter_by_coordinate_system('does_not_exist') -> elements:", list(out.gen_elements()))
    print("VERDICT: REPRODUCED (empty SpatialData returned, no error/warning for a non-existing coordinate system)")
except Exception as e:  # noqa: BLE001
    print("raised:", type(e).__name__, e)
    print("VERDICT: NOT REPRODUCED (an error is raised)")
```

**Environment:** `uv run --no-project --python 3.13`, which resolves `spatialdata` from `git+https://github.com/scverse/spatialdata.git@ccf1ea048d054b6624214bf618008a9f9ae223e0` (extra dependencies, if any, are pinned in the script's inline metadata above).

**Output** (verbatim, only `uv` package-resolution/install log lines removed):

```
coordinate systems: ['global']
filter_by_coordinate_system('does_not_exist') -> elements: []
VERDICT: REPRODUCED (empty SpatialData returned, no error/warning for a non-existing coordinate system)
exit=0
```

**Agent's severity assessment:** low (see the `priority: *` label added to this issue).

---
_Automatically generated; discovered by an AI agent (Claude) and not yet reviewed by a human._
